### ISSUE TITLE
Serial Terminal Glitches

### ISSUE THREAD

**Issue by csavage-parallax** [parallaxinc/BlocklyProp#836]
You can remove this if it is known or otherwise doesn't belong here, but I accidentally right-clicked in the serial terminal window and got the following results (first image is before, second is after).

![blockly_serial console_01](https://cloud.githubusercontent.com/assets/20691193/20844460/c0fa2c80-b874-11e6-80c8-f2bebe77650f.jpg)

![blockly_serial console_02](https://cloud.githubusercontent.com/assets/20691193/20844463/c644cf4c-b874-11e6-9d1d-6b936bc19ba6.jpg)

**Comment by zfi** [comment]
@csavage-parallax: Can this error be replicated? Does this happen with just a specific project or multiple projects?

**Comment by csavage-parallax** [comment]
I will try some more tests to determine. I noticed it on the one I posted, so I mentioned it. I will follow up with some simpler test projects to demonstrate this.

**Comment by csavage-parallax** [comment]
Glitch happens in a new simple project as well: http://demo.blockly.parallax.com/blockly/my/projects.jsp#481

![image](https://cloud.githubusercontent.com/assets/20691193/21112986/b8c115a2-c05c-11e6-98fe-b0fc969797ee.png)

**Comment by MatzElectronics** [comment]
Another glitch that needs to be addressed:
![image](https://cloud.githubusercontent.com/assets/19737272/21205481/56250a82-c211-11e6-8de1-49052f959f9f.png)

**Comment by MatzElectronics** [comment]
@zfi @PropGit @kgracey @csavage-parallax I can replicate the right click behavior as described above.  The over-extension of the scroll bar is something that has always been there (discovered by looking at old screenshots) and can hopefully be fixed with a tweak to the CSS.

The terminal is a set of divs - one for each line to allow cursor positioning.  That means that there are a fixed number of lines vertically and it won't ever be possible to scroll up through past data.

**I propose disabling right-clicking in the terminal**.  This means that an individual will not be able to select and copy anything, so **I also propose adding a "copy to clipboard" button at the bottom**.  Also, it may be worth looking into a disable button that allows the terminal to stay open but stop updating so the user can "freeze" the screen.

Basically - this is NOT a full featured terminal, and we need to perhaps define and stick with its limitations.

**Comment by PropGit** [comment]
@MatzElectronics - Wait... 
> That means that there are a fixed number of lines vertically and it won't ever be possible to scroll up through past data.
Has the terminal been changed to behave this way?  It was designed to allow scrolling up and down to see previous/next results... but the vertical scroll bar has always been missing (I believe) as I reported in another issue long ago; the mouse-wheel and also shift (or control?)+mouse-wheel scrolls the display in all four directions.  There are also still outstanding issues regarding terminal behavior in relation to cursor placement and new data arrival in relation to scroll position.

I agree to disabling right-clicking and adding a copy to clipboard button.

**Comment by MatzElectronics** [comment]
To be honest - I don't know the history, but it doesn't look like the
terminal in it's current form is capable of scrolling up.  I think that it
is the way that it is to enable cursor positioning.

That being said - if we create a "copy to clipboard", we could store
everything disappearing upward and collect it in a buffer to attach back to
what gets sent to the clipboard.

*Matthew Matz  |  STEM/Robotics Educator*
Parallax Inc.  |  Direct: 916-625-3019  |  www.parallax.com  |  @M
atzElectronics <http://twitter.com/MatzElectronics>

**Comment by MatzElectronics** [comment]
Regardless - the terminal is a beast...

*Matthew Matz  |  STEM/Robotics Educator*
Parallax Inc.  |  Direct: 916-625-3019  |  www.parallax.com  |  @M
atzElectronics <http://twitter.com/MatzElectronics>
